test/sqllogictest/funcs.slt

```diff
@@ -533,3 +558,13 @@ query I
 SELECT array_lower(ARRAY[[['a', 'b']]], 4)
 ----
 NULL
+
+query I
+SELECT array_upper(ARRAY[[[1, 2]]], 3)
+----
+2
+
+query I
+SELECT array_upper(ARRAY[[[1, 2]]], 4)
+----
+NULL
```

Comment:
Just for completion's sake, can we add some tests that use _NULL_ values in the input? e.g. 

```
query I
SELECT array_upper(NULL, 1)

query I
SELECT array_upper(ARRAY[NULL], 1)

query I
SELECT array_upper(ARRAY[1], NULL)
```